### PR TITLE
add project

### PR DESCRIPTION
**Pull request by AmMozart**

### CODE REVIEW

File: homeworks/03-data-structure/src/binary-tree/BinaryTree.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  private traverseDFS(treeNode: TreeNode<T>, traverseType: TraverseType): T[] {
+    const values: T[] = []
+
+    if (treeNode) {
+      traverseType === TraverseType.PreOrder && values.push(treeNode.value)
+      values.push(...this.traverseDFS(treeNode.left, traverseType))
+      traverseType === TraverseType.InOrder && values.push(treeNode.value)
+      values.push(...this.traverseDFS(treeNode.right, traverseType))
+      traverseType === TraverseType.PostOrder && values.push(treeNode.value)
+    }
```
Comment (shpakkdv) [inline homeworks/03-data-structure/src/binary-tree/BinaryTree.ts:77]: Seems, switch/case is more appropriate here.

File: homeworks/03-data-structure/src/binary-tree/BinaryTree.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    return values
+  }
+
+  private traverseBFS(treeNode: TreeNode<T>): T[] {
+    const values: T[] = []
+    const treeNodeQueue = [treeNode]
+
+    while (treeNodeQueue.length) {
+      const dot = treeNodeQueue.shift()
```
Comment (shpakkdv) [inline homeworks/03-data-structure/src/binary-tree/BinaryTree.ts:87]: `dot`? What does it mean?

File: homeworks/03-data-structure/src/tests/BinarySearchTree.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { BinarySearchTree } from "../binary-tree/BinarySearchTree"
+import { TreeNode } from "../binary-tree/TreeNode"
+
+const testTree = require('./testTree')
+
+describe("BinarySearchTree Module:", () => {
```
Comment (shpakkdv) [inline homeworks/03-data-structure/src/tests/BinarySearchTree.test.ts:6]: I would expect tests of inherited methods as well.

**Review by shpakkdv — APPROVED** [review]
Looks good 👍

**Comment by AmMozart** [comment]
end project

**Pull request merged**